internal/orchestration/controller.go

```diff
@@ -342,14 +343,31 @@ func (e *Experiment) stopReplicas() error {
 		for id, hash := range res.GetHashes() {
 			hashes[id] = hash
 		}
+		for id, cnt := range res.GetCounts() {
+			counts[id] = cnt
+		}
+	}
+	for id1, cnt1 := range counts {
+		for id2, cnt2 := range counts {
+			if id2 > id1 && cnt2 == cnt1 {
+				if !bytes.Equal(hashes[id1], hashes[id2]) {
+					return fmt.Errorf("hash mismatch for same count")
```

Comment:
Not sure if it is necessary or interesting, but would it make sense to include the mismatching IDs in the error message?